flink-rpc/flink-rpc-core/src/main/java/org/apache/flink/runtime/rpc/RpcSystem.java

```diff
@@ -91,8 +93,11 @@ static RpcSystem load() {
      * @return loaded RpcSystem
      */
     static RpcSystem load(Configuration config) {
```

Comment:
Could we add a unit test for that checks that the right loader is picked based on the priority?

Reply:
kinda? We could define a custom one that uses -1, but technically it violates the contract (since 0 is MAX).

Reply:
We can of course set the main priority to 1 instead and reserve 0 for testing or something